src/tree/ProxiesTreeItem.ts

```diff
@@ -0,0 +1,101 @@
+
+    public constructor(siteWrapper: SiteWrapper, outputChannel: vscode.OutputChannel) {
+        this._siteWrapper = siteWrapper;
+        this._outputChannel = outputChannel;
+    }
+
+    public get id(): string {
+        return 'proxies';
+    }
+
+    public get iconPath(): nodeUtils.IThemedIconPath {
+        return nodeUtils.getThemedIconPath('BulletList');
+    }
+
+    public hasMoreChildren(): boolean {
+        return false;
+    }
+
+    public async loadMoreChildren(node: IAzureNode<IAzureTreeItem>, _clearCache: boolean | undefined): Promise<IAzureTreeItem[]> {
+        let proxiesJson: string;
+        try {
+            const client: KuduClient = await nodeUtils.getKuduClient(node, this._siteWrapper);
+            const result: IFileResult = await this._siteWrapper.getFile(client, this._proxiesJsonPath);
+            proxiesJson = result.data;
+            this._etag = result.etag;
+        } catch (err) {
+            // if the proxies.json file does not exist, that means there are no proxies
+            return [];
+        }
+
+        try {
+            const rawProxyConfig: IRawProxyConfig = <IRawProxyConfig>JSON.parse(proxiesJson);
+            if (!rawProxyConfig.proxies) {
+                rawProxyConfig.proxies = {};
+            }
+            this._proxyConfig = <IProxyConfig>rawProxyConfig;
+            return Object.keys(this._proxyConfig.proxies).map((name: string) => new ProxyTreeItem(name));
+        } catch (err) {
+            throw new Error(localize('failedToParseProxyConfig', 'Failed to parse "proxies.json" file: {0}', parseError(err).message));
+        }
+    }
+
+    public async deleteProxy(node: IAzureNode, name: string): Promise<void> {
+        const message: string = localize('azFunc.ConfirmDelete', 'Are you sure you want to delete proxy "{0}"?', name);
+        if (await vscode.window.showWarningMessage(message, DialogResponses.yes, DialogResponses.cancel) === DialogResponses.yes) {
+            if (this._deletingProxy) {
+                throw new Error(localize('multipleProxyOperations', 'An operation on the proxy config is already in progress. Wait until it has finished and try again.'));
+            } else {
+                this._deletingProxy = true;
+                try {
+                    this._outputChannel.show(true);
+                    this._outputChannel.appendLine(localize('DeletingProxy', 'Deleting proxy "{0}"...', name));
+                    delete this._proxyConfig.proxies[name];
+                    const client: KuduClient = await nodeUtils.getKuduClient(node, this._siteWrapper);
+                    const data: string = JSON.stringify(this._proxyConfig);
+                    this._etag = await this._siteWrapper.putFile(client, data, this._proxiesJsonPath, this._etag);
+                    this._outputChannel.appendLine(localize('DeleteProxySucceeded', 'Successfully deleted proxy "{0}".', name));
+                } finally {
```

Comment:
With no catch statement, the error will get swallowed if an exception occurs during the delete, right?

Reply:
No - the exception will not get swallowed. This try/finally has nothing to do with error handling and everything to do with the 'deletingProxy' flag. You need to make sure the flag is _always_ set back to the original value, otherwise a failure will block the user from ever performing the operation again